utils/platform/users.go

```diff
@@ -0,0 +1,160 @@
+package platform
+
+import (
+	"context"
+	"fmt"
+	"os/user"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/parallelisation"
+)
+
+// DefineUser adds a new user to the platform
+func DefineUser(ctx context.Context, user *user.User, password string) (err error) {
+	if user == nil {
+		err = fmt.Errorf("%w: missing user information", commonerrors.ErrUndefined)
+		return
+	}
+	username := user.Username
+	if username == "" {
+		username = user.Uid
+	}
+	err = AddUser(ctx, username, user.Name, password)
+	return
+}
+
+// AddUser adds a new user to the platform
+func AddUser(ctx context.Context, username, fullname, password string) error {
+	err := parallelisation.DetermineContextError(ctx)
+	if err != nil {
+		return err
+	}
+	found, _ := HasUser(username)
+	if found {
+		return nil
+	}
+	return ConvertUserGroupError(addUser(ctx, username, fullname, password))
+}
+
+// DeleteUser removes a user from the platform
```

Comment:
Can there be more of a distinction between when to use remove Vs delete user? If I was importing this library I wouldn't know which to use

Reply:
tried to do so